source/common/upstream/load_balancer_impl.cc

```diff
@@ -236,22 +236,67 @@ void LoadBalancerBase::recalculatePerPriorityPanic() {
   const uint64_t panic_threshold = std::min<uint64_t>(
       100, runtime_.snapshot().getInteger(RuntimePanicThreshold, default_healthy_panic_percent_));
 
-  // Panic mode is disabled only when panic_threshold is 0%.
-  if (panic_threshold > 0 && normalized_total_availability == 0) {
-    // Everything is terrible. All load should be to P=0. Turn on panic mode.
-    ASSERT(per_priority_load_.healthy_priority_load_.get()[0] == 100);
-    per_priority_panic_[0] = true;
+  // This is corner case when panic is disabled and there is no hosts available.
+  // But the load must be distributed somewhere.
+  // This is just a placeholder, because no traffic will be directed to
+  // priority 0 anyways, because it is empty.
```

Comment:
technically its either empty or has no healthy hosts, right? would be good to be specific

Reply:
Yeah, this comment doesn't seem quite right.  P[0] may have hosts (it's not necessarily empty).  In theory it would just fail to accept queries right?

Reply:
Also I think there's a comment above recalculatePerPriorityState which I think may need updating.

Reply:
Updated the comment above recalculatePerPriorityState. Added explanation why 100% of load is assigned to P=0 when there are no healthy hosts and panic is disabled. Other routines expect that the sum of all percentages adds to 100%. If this is not the case there are assertions which fail. When panic is disabled, the load balancers will look for healthy hosts and because there is none, error will be returned and nothing will be routed to backend cluster.